samvel-aivazian/lesson6-hw2-5.py

```diff
@@ -0,0 +1,98 @@
+import sys
+
+
+# Assignment 2: The Substring Counter
+def count_substrings(main_string, substring):
+    try:
+        return main_string.count(substring)
+    except Exception as e:
+        print(f"An error occurred: {e}", file=sys.stderr)
+
+
+main_string_test = "hellohellohello"
+substring_test = "ello"
+
+try:
+    substring_count = count_substrings(main_string_test, substring_test)
+    print(substring_count, f"The substring '{substring_test}' appears {substring_count} times in '{main_string_test}'.")
+except Exception as e:
+    print(f"An error occurred: {e}", file=sys.stderr)
+
+
+# Assignment 3: The Alignment Formatter
+def format_sentence(sentence, frame_width):
+    try:
+        if frame_width < len(sentence):
+            raise ValueError("Frame width is too small for the sentence.")
+        return sentence.center(frame_width)
+    except Exception as e:
+        print(f"An error occurred: {e}", file=sys.stderr)
+
+
+sentence_test = "hello"
+frame_width_test = 10
+
+try:
+    formatted_sentence = format_sentence(sentence_test, frame_width_test)
+    print(formatted_sentence)
+except Exception as e:
+    print(f"An error occurred: {e}", file=sys.stderr)
+
+
+# Assignment 4: The Case Converter
+def convert_case(text, case):
+    try:
+        if case.lower() == 'upper':
+            return text.upper()
+        elif case.lower() == 'lower':
+            return text.lower()
+        else:
+            raise ValueError("Case must be 'upper' or 'lower'.")
+    except Exception as e:
+        print(f"An error occurred: {e}", file=sys.stderr)
```

Comment:
This is totally wrong, catching the `except Exception as e:` and raise `ValueError("Case must be 'upper' or 'lower'.")` 

Handle it with an exception chaining